src/timepicker/timepicker.js

```diff
@@ -165,7 +165,7 @@ angular.module('ui.bootstrap.timepicker', [])
     var hours = +$scope.hours;
     var valid = $scope.showMeridian ? hours > 0 && hours < 13 :
       hours >= 0 && hours < 24;
-    if (!valid) {
+    if (!valid || $scope.hours === '') {
```

Comment:
This is fine - `valid` is a boolean, as can be seen from the two lines prior. I did do a double take too, but it checks out.